File: leetcode/1.cpp
#include "base.hpp"
#include <unordered_map>

using ll = long long;
using vi = vector<int>;
using vii = vector<vector<int>>;

class Solution {
public:
  vector<int> twoSum(vector<int> &nums, int target) {
    unordered_map<int, int> mp;
    for (int i{}, _ = nums.size(); i < _; ++i) {
      mp[nums[i]] = i;
    }
    for (int i{}, _ = nums.size(); i < _; ++i) {
      if (auto itr = mp.find(target - nums[i]); itr != mp.end()) {
        if (i != itr->second)
          return {i, itr->second};
      }
    }
    return {};
  }
};

int main() {}

File: leetcode/1.py
nums = []
for i in range(1, 100000):
    p1 = int(str(i)+str(i)[::-1])**2
    p2 = int(str(i)[:-1]+str(i)[::-1])**2
    if(str(p1) == str(p1)[::-1]):
        nums.append(p1)
    if(str(p2) == str(p2)[::-1]):
        nums.append(p2)

print(nums)

ans = []

for i in nums:
    if((len(str(i))) % 2 == 0):
        ans.append(i)

print(ans)
